src/webhook/mutation/pod_mutator/init_container.go

```diff
@@ -44,34 +43,59 @@ func securityContextForInitContainer(pod *corev1.Pod) *corev1.SecurityContext {
 				"ALL",
 			},
 		},
-		SeccompProfile: &corev1.SeccompProfile{
-			Type: corev1.SeccompProfileTypeRuntimeDefault,
-		},
 	}
 
+	copySecurityContextUserAndGroup(securityContext, *pod)
+
+	return securityContext
+}
+
+func copySecurityContextUserAndGroup(securityContext *corev1.SecurityContext, pod corev1.Pod) {
 	containerSecurityContext := pod.Spec.Containers[0].SecurityContext
 	podSecurityContext := pod.Spec.SecurityContext
 
-	if hasContainerUserAndGroupSet(containerSecurityContext) {
-		securityContext.RunAsGroup = containerSecurityContext.RunAsGroup
-		securityContext.RunAsUser = containerSecurityContext.RunAsUser
-	} else if hasPodUserAndGroupSet(podSecurityContext) {
-		securityContext.RunAsGroup = podSecurityContext.RunAsGroup
+	securityContext.RunAsUser = address.Of(defaultUser)
+	securityContext.RunAsGroup = address.Of(defaultGroup)
+
+	if hasPodUserSet(podSecurityContext) {
```

Comment:
should we even set user/group for the init container when the pod one would still be valid here?

Reply:
this is so the code is kept simple (IMO)

otherwise to determine if we can set the `RunAsNonRoot` we would have to check if we set anything in our securityContext, and if not then check the pod securityContext

so the current logic is to set the securityContext to default then overwrite it in revers order of the priority of the user's securityContext (Priority 1. Container SC; Priority 2. Pod SC; Priority 3. Default SC) , then check the resulting securityContext if its root or not